diff --git a/node_modules/@qwik-ui/headless/components/modal/index.d.ts b/node_modules/@qwik-ui/headless/components/modal/index.d.ts
index 8d1e78a..0227b67 100644
--- a/node_modules/@qwik-ui/headless/components/modal/index.d.ts
+++ b/node_modules/@qwik-ui/headless/components/modal/index.d.ts
@@ -7,3 +7,4 @@ export { HModalTitle as Title } from './modal-title';
 export { HModalDescription as Description } from './modal-description';
 export { HModalTrigger as Trigger } from './modal-trigger';
 export { HModalClose as Close } from './modal-close';
+export { modalContextId as contextId } from "./modal-context";
diff --git a/node_modules/@qwik-ui/headless/components/modal/index.qwik.cjs b/node_modules/@qwik-ui/headless/components/modal/index.qwik.cjs
index 3fcde87..8ad5b71 100644
--- a/node_modules/@qwik-ui/headless/components/modal/index.qwik.cjs
+++ b/node_modules/@qwik-ui/headless/components/modal/index.qwik.cjs
@@ -9,6 +9,7 @@ const modalTitle = require("./modal-title.qwik.cjs");
 const modalDescription = require("./modal-description.qwik.cjs");
 const modalTrigger = require("./modal-trigger.qwik.cjs");
 const modalClose = require("./modal-close.qwik.cjs");
+const modalContext = require("./modal-context.qwik.cjs");
 exports.Root = modalRoot.HModalRoot;
 exports.Panel = modalPanel.HModalPanel;
 exports.Content = modalContent.HModalContent;
@@ -18,3 +19,4 @@ exports.Title = modalTitle.HModalTitle;
 exports.Description = modalDescription.HModalDescription;
 exports.Trigger = modalTrigger.HModalTrigger;
 exports.Close = modalClose.HModalClose;
+exports.contextId = modalContext.modalContextId;
diff --git a/node_modules/@qwik-ui/headless/components/modal/index.qwik.mjs b/node_modules/@qwik-ui/headless/components/modal/index.qwik.mjs
index bc3cfdb..2f71c9a 100644
--- a/node_modules/@qwik-ui/headless/components/modal/index.qwik.mjs
+++ b/node_modules/@qwik-ui/headless/components/modal/index.qwik.mjs
@@ -7,6 +7,7 @@ import { HModalTitle } from "./modal-title.qwik.mjs";
 import { HModalDescription } from "./modal-description.qwik.mjs";
 import { HModalTrigger } from "./modal-trigger.qwik.mjs";
 import { HModalClose } from "./modal-close.qwik.mjs";
+import { modalContextId } from "./modal-context.qwik.mjs";
 export {
   HModalClose as Close,
   HModalContent as Content,
@@ -16,5 +17,6 @@ export {
   HModalPanel as Panel,
   HModalRoot as Root,
   HModalTitle as Title,
-  HModalTrigger as Trigger
+  HModalTrigger as Trigger,
+  modalContextId as contextId
 };
